buildstockbatch/gcp/main.tf

```diff
@@ -0,0 +1,73 @@
+# Terraform file to define GCP resources used by buildstock_gcp
+#
+# When setting up a new project:
+#   terraform init
+#
+# To see what changes will be applied:
+#   terraform plan
+#
+# To apply those changes:
+#   terraform apply
+#
+# Optionally set variables:
+#   terraform apply -var="gcp_project=myproject" -var="bucket_name=mybucket" -var="region=us-east1-b"
+#
+# Format this file:
+#   terraform fmt main.tf
+
+
+variable "gcp_project" {
+  type        = string
+  default     = "buildstockbatch-dev"
+  description = "GCP project to use"
+}
+
+variable "bucket_name" {
+  type        = string
+  default     = "buildstockbatch"
+  description = "GCS bucket where buildstockbatch inputs and outputs should be stored"
+}
+
+variable "topic_name" {
+  type        = string
+  default     = "notifications"
+  description = "Pub/sub topic where batch job state change notifications will be sent"
+}
+
+variable "region" {
+  type        = string
+  default     = "us-central1"
+  description = "GCP region where all resources will be created"
+}
+
+variable "artifact_registry_repository" {
+  type        = string
+  default     = "buildstockbatch-docker"
+  description = "Name of the artifact registry repository for storing Docker images. May contain letters, numbers, hyphens."
+}
+
+
+provider "google" {
+  project = var.gcp_project
+  region  = var.region
+}
+
+# Pub/sub topic for job notifications
+resource "google_pubsub_topic" "notification_topic" {
+  name = var.topic_name
+}
+
+# GCS bucket for storing inputs and outputs
```

Comment:
Something to consider (for later work, probably): A different bucket for intermediate files with a different storage class. Or, is there a way—with Terraform—to somehow configure some files/paths within a bucket to have a different storage class?

Reply:
Won't everything probably use the [Standard](https://cloud.google.com/storage/docs/storage-classes) storage class? Or do you mean [location type](https://cloud.google.com/storage/docs/locations#considerations)?

Storage classes can be set per-file, but location type is for the whole bucket.

Reply:
Haven't actually thought it all the way through 😅...

But basically, `results/parquet` is all that we care about after running simulations (where "after" might be days or weeks later). So, it could make sense to reduce the cost of other files (basically what's in `results/simulation_output`), which can be accomplished through Lifecycle Management (automatically delete such files after some time), and/or non-Standard Storage Class. I would guess that Regional for location type is good for the whole bucket.